src/validation/custom.validation.ts

```diff
@@ -78,11 +78,75 @@ export const checkDateValueIsValid = (errMsg: string, dayStr: string = "", month
 
 export const checkOptionalDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
   if ( dayStr !== "" || monthStr !== "" || yearStr !== "" ) {
-    checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
-    checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+    const areDateFieldsPresent = checkDateFieldsPresent(dayStr, monthStr, yearStr);
+    if (areDateFieldsPresent) {
+      const isDateValid = checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
+      if (isDateValid) {
+        const isDateInThePast = checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+        if (isDateInThePast) {
+          checkDateIsWithinLast3Months(ErrorMessages.IDENTITY_CHECK_DATE_NOT_WITHIN_PAST_3_MONTHS, dayStr, monthStr, yearStr);
+        }
+      }
+    }
+  }
+  return true;
+};
+
+export const checkIdentityDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  const isStartDateFields = checkStartDate(dayStr, monthStr, yearStr);
+  if (isStartDateFields) {
     checkDateIsWithinLast3Months(ErrorMessages.IDENTITY_CHECK_DATE_NOT_WITHIN_PAST_3_MONTHS, dayStr, monthStr, yearStr);
   }
+  return true;
+};
 
+export const checkStartDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  const isDatePresent = checkDateIsNotCompletelyEmpty(ErrorMessages.ENTER_DATE, dayStr, monthStr, yearStr);
+  if (isDatePresent) {
+    const areDateFieldsPresent = checkDateFieldsPresent(dayStr, monthStr, yearStr);
+    if (areDateFieldsPresent) {
+      const isDateValid = checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
+      if (isDateValid) {
+        checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+      }
+    }
+  }
+  return true;
+};
+
+export const checkDateFieldsPresent = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
```

Comment:
Look at the naming of the other methods in this file - they have 'Is' within the function name, so to be consistent, suggest this is renamed to `checkDateFieldsArePresent`.

Reply:
Renamed as above